app/controllers/ConsentController.java

```diff
@@ -81,15 +81,13 @@ private Result changeSharingScope(SharingScope sharingScope, String message) {
     }
 
     private Result giveConsentForVersion(int version) throws Exception {
+
         final UserSession session = getAuthenticatedSession();
         final Study study = studyService.getStudy(session.getStudyIdentifier());
         final JsonNode node = requestToJSON(request());
-        
         final ConsentSignature consent = ConsentSignature.createFromJson(node);
-        final User user = consentService.consentToResearch(study, session.getUser(), consent, true);
-        
-        SharingOption sharing = SharingOption.fromJson(node, version);
-        optionsService.setOption(study, session.getUser().getHealthCode(), sharing.getSharingScope());
+        final SharingOption sharing = SharingOption.fromJson(node, version);
+        final User user = consentService.consentToResearch(study, session.getUser(), consent, sharing, true);
```

Comment:
Let's pass the enumeration to the service, leaving SharingOption as a form in the controller.